webv2/primary.go

```diff
@@ -0,0 +1,319 @@
+package webv2
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+	"github.com/cloudspannerecosystem/harbourbridge/webv2/session"
+)
+
+// PrimaryKeyRequest represents  Primary keys API Payload
+type PrimaryKeyRequest struct {
+	TableId      int      `json:"TableId"`
+	Columns      []Column `json:"Columns"`
+	PrimaryKeyId int      `json:"PrimaryKeyId"`
+}
+
+/*PrimaryKeyResponse represents  Primary keys API response
+Synth is true is for table Primary Key Id is not present and it is generated
+*/
+type PrimaryKeyResponse struct {
+	TableId      int      `json:"TableId"`
+	Columns      []Column `json:"Columns"`
+	PrimaryKeyId int      `json:"PrimaryKeyId"`
+	Synth        bool     `json:"Synth"`
+}
+
+type Column struct {
+	ColumnId int    `json:"ColumnId"`
+	ColName  string `json:"ColName"`
+	Desc     bool   `json:"Desc"`
+	Order    int    `json:"Order"`
+}
+
+//primaryKey updates Primary keys in Spanner Table.
+func primaryKey(w http.ResponseWriter, r *http.Request) {
+
+	reqBody, err := ioutil.ReadAll(r.Body)
+
+	if err != nil {
+		log.Println(err)
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+	}
+
+	pkRequest := PrimaryKeyRequest{}
+
+	err = json.Unmarshal(reqBody, &pkRequest)
+	if err != nil {
+		log.Println(err)
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+
+	sessionState := session.GetSessionState()
+	spannerTable, found := getSpannerTable(sessionState, pkRequest)
+
+	if !found {
+		log.Println(err)
+		http.Error(w, fmt.Sprintf("tableId not found : %v", err), http.StatusNotFound)
+		return
+
+	}
+
+	if len(pkRequest.Columns) == 0 {
+		log.Println(err)
+		http.Error(w, fmt.Sprintf("empty columm error : %v", err), http.StatusBadRequest)
+		return
+
+	}
+
+	if !isValidColumnIds(pkRequest, spannerTable) {
+		log.Println(err)
+		http.Error(w, fmt.Sprintf("colummId not found error : %v", err), http.StatusBadRequest)
+		return
+
+	}
+
+	updatePrimaryKey(pkRequest, spannerTable)
+	spannerTable = insertOrRemovePrimarykey(pkRequest, spannerTable)
+	pKeyResponse := prepareResponse(pkRequest, spannerTable)
+
+	for _, table := range sessionState.Conv.SpSchema {
+		if pkRequest.TableId == table.Id {
+			sessionState.Conv.SpSchema[table.Name] = spannerTable
+		}
+	}
+
+	updateSessionFile()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(pKeyResponse)
+}
+
+func getSpannerTable(sessionState *session.SessionState, pkRequest PrimaryKeyRequest) (spannerTable ddl.CreateTable, found bool) {
```

Comment:
These do not seem like PrimaryKey specific methods. They should probably be in a common module/file